edb/common/topological.py

```diff
@@ -44,6 +44,7 @@ class DepGraphEntry(Generic[K, V, T]):
 
     item: V
     deps: MutableSet[K]
+    weak_deps: MutableSet[K]
```

Comment:
Could you add some comments on what the semantics of weak_deps are and how they are implemented?